src/main/java/net/william278/velocitab/packet/ScoreboardManager.java

```diff
@@ -85,27 +90,57 @@ private void dispatchPacket(@NotNull UpdateTeamsPacket packet, @NotNull Player p
             plugin.getTabList().removeOfflinePlayer(player);
             return;
         }
+
         try {
-            ProtocolizePlayer protocolizePlayer = Protocolize.playerProvider().player(player.getUniqueId());
-            if (protocolizePlayer != null) {
-                protocolizePlayer.sendPacket(packet);
-            } else {
-                plugin.log("Failed to get ProtocolizePlayer for player " + player.getUsername() + " (UUID: " + player.getUniqueId() + ")");
-            }
+            final ConnectedPlayer connectedPlayer = (ConnectedPlayer) player;
+            connectedPlayer.getConnection().write(packet);
         } catch (Exception e) {
             plugin.log("Failed to dispatch packet (is the client or server modded or using an illegal version?)", e);
         }
     }
 
+    private static final MethodHandle STATE_REGISTRY$clientBound;
+    private static final MethodHandle PACKET_REGISTRY$register;
+    private static final MethodHandle PACKET_MAPPING$map;
+
+    static {
+        final MethodHandles.Lookup lookup = MethodHandles.lookup();
+        try {
+            final MethodHandles.Lookup stateRegistryLookup = MethodHandles.privateLookupIn(StateRegistry.class, lookup);
+            STATE_REGISTRY$clientBound = stateRegistryLookup.findGetter(StateRegistry.class, "clientbound", StateRegistry.PacketRegistry.class);
+
+            final MethodType mapType = MethodType.methodType(StateRegistry.PacketMapping.class, Integer.TYPE, ProtocolVersion.class, Boolean.TYPE);
+            PACKET_MAPPING$map = stateRegistryLookup.findStatic(StateRegistry.class, "map", mapType);
+
+            final MethodHandles.Lookup packetRegistryLookup = MethodHandles.privateLookupIn(StateRegistry.PacketRegistry.class, lookup);
+            final MethodType registerType = MethodType.methodType(void.class, Class.class, Supplier.class, StateRegistry.PacketMapping[].class);
+            PACKET_REGISTRY$register = packetRegistryLookup.findVirtual(StateRegistry.PacketRegistry.class, "register", registerType);
+        } catch (Throwable e) {
```

Comment:
This reminds me that we should probably catch Throwable when writing the packet, too, instead of Exception as we're currently doing.